lizmap/modules/lizmap/classes/qgisExpressionUtils.class.php

```diff
@@ -0,0 +1,282 @@
+            );
+            if ($form_feature) {
+                $params['feature'] = json_encode($form_feature);
+                $params['form_scope'] = 'true';
+            }
+
+            // Request evaluate expression
+            $json = self::request($params);
+            if (!$json) {
+                return null;
+            }
+            if (property_exists($json, 'status') && $json->status != 'success') {
+                // TODO parse errors
+                // if (property_exists($json, 'errors')) {
+                // }
+                jLog::log($data, 'error');
+            } else if (property_exists($json, 'results') &&
+                array_key_exists(0, $json->results)) {
+                // Get results
+                return $json->results[0];
+            } else {
+                // Data not well formed
+                jLog::log($data, 'error');
+            }
+        }
+        return null;
+    }
+
+    static public function getFeatureWithFormScope($layer, $expression, $form_feature, $fields)
+    {
+
+        $project = $layer->getProject();
+        $plugins = $project->getQgisServerPlugins();
+        if (array_key_exists('Lizmap', $plugins)) {
+            // build parameters
+            $params = array(
+                'service' => 'EXPRESSION',
+                'request' => 'getFeatureWithFormScope',
+                'map' => $project->getRelativeQgisPath(),
+                'layer' => $layer->getName(),
+                'filter' =>  self::updateExpressionByUser($layer, $expression),
+                'form_feature' => json_encode($form_feature),
+                'fields' => implode(',', $fields)
+            );
+
+            // Request getFeatureWithFormsScope
+            $json = self::request($params);
+            if (!$json || !property_exists($json, 'features')) {
+                return array();
+            }
+
+            return $json->features;
+        }
+
+        return array();
+    }
+
+    /**
+     * Return form group visibilities.
+     *
+     * @param qgisAttributeEditorElement $attributeEditorForm
+     * @param jFormsBase                 $form
+     *
+     * @return array Visibilities, an associated array with group html id as key and boolean as value.
+     *
+     */
+    static public function evaluateGroupVisibilities($attributeEditorForm, $form)
+    {
+        // qgisForm::getAttributesEditorForm can return null
+        if ($attributeEditorForm === null || $form === null) {
+            return array();
+        }
+        // Get criterias and expressions to evaluate
+        // and prepare visibilities
+        $criteriaFrom = array();
+        $expressions = array();
+        $visibilities = array();
+        $visibilityExpressions = $attributeEditorForm->getGroupVisibilityExpressions();
+        foreach ($visibilityExpressions as $id => $exp) {
+            $visibilities[$id] = True;
+
+            if ($exp === null || $exp === '') {
```

Comment:
same question as above if `$exp` has already been trimed before this test

Reply:
fixed